src/popup/App.vue

```diff
@@ -163,9 +170,18 @@ export default defineComponent({
 
       await watchUntilTruthy(() => isRestored.value);
       setNotificationSettings();
+      startListeningForIncomingTransactions();
 
       loadAeternityData();
       root.$store.commit('setChainNames', await root.$store.dispatch('getCacheChainNames'));
+
+      if (IS_WEB) {
+        webNotification.requestPermission();
+      }
```

Comment:
In my opinion this piece of code should be put closer to the code that actually displays the notification to the user. I see few solutions for this:
1. In the `pushNotification` file instead of having functions create a singleton JS class with the public method that displays the notification and the constructor asks for the permission.
2. Encapsulate both asking for permission and sending the notification in the `notifications.ts` composable.

If I may choose I would go with the 1st one as it follows similar pattern as the `Logger`.

Reply:
if you request notification permission while the user is idle in the app the browser will automatically reject it.
cause browsers expects permission request after user click or page reload

Reply:
OK, so the singleton could have the `init` method which could be run in the App.vue file.
What I want is to have a clear abstraction encapsulated in one file that handles all the logic related with native notifications.